libraries/AP_Networking/AP_Networking_port.cpp

```diff
@@ -368,23 +384,25 @@ bool AP_Networking::Port::send_receive(void)
             WITH_SEMAPHORE(sem);
             n = writebuffer->peekbytes(buf, available);
         }
-        if (n > 0) {
-            const auto ret = sock->send(buf, n);
-            if (ret > 0) {
-                WITH_SEMAPHORE(sem);
-                writebuffer->advance(ret);
-                active = true;
-            }
+
+        // nothing to send return
+        if (n <= 0) {
+            return active;
         }
-    } else {
-        if (type == NetworkPortType::UDP_SERVER && have_received) {
-            // connect the socket to the last receive address if we have one
-            char buf[16];
-            uint16_t last_port;
-            const char *last_addr = sock->last_recv_address(buf, sizeof(buf), last_port);
-            if (last_addr != nullptr && port != 0) {
-                connected = sock->connect(last_addr, last_port);
-            }
+
+        ssize_t ret = -1;
+        if (type == NetworkPortType::UDP_SERVER &&
+            last_udp_connect_address != 0 &&
+            last_udp_connect_port != 0) {
+            ret = sock->sendto(buf, n, last_udp_connect_address, last_udp_connect_port);
+        } else {
+            ret = sock->send(buf, n);
```

Comment:
this send makes no sense without having received a pkt when we are a UDP server

Reply:
made the condition more tied in with non-UDP_SERVER sends. And added comments for more readability